pkg/v1/client.go

```diff
@@ -1 +1,201 @@
+	defaultTLSHandshakeTimeout = 60
+
+	// defaultExpectContinueTimeout represents the default amount of time to wait for a server's first
+	// response headers.
+	defaultExpectContinueTimeout = 1
+)
+
+// ServiceClient stores details that are needed to work with Selectel Managed Kubernetes Service API.
+type ServiceClient struct {
+	// HTTPClient represents an initialized HTTP client that will be used to do requests.
+	HTTPClient *http.Client
+
+	// TokenID is a client authentication token.
+	TokenID string
+
+	// Endpoint represents an endpoint that will be used in all requests.
+	Endpoint string
+
+	// UserAgent contains user agent that will be used in all requests.
+	UserAgent string
+}
+
+// NewMKSClientV1 initializes a new MKS client for the V1 API.
+func NewMKSClientV1(tokenID, endpoint string) *ServiceClient {
+	return &ServiceClient{
+		HTTPClient: newHTTPClient(),
+		TokenID:    tokenID,
+		Endpoint:   endpoint,
+		UserAgent:  userAgent,
+	}
+}
+
+// newHTTPClient returns a reference to an initialized and configured HTTP client.
+func newHTTPClient() *http.Client {
+	return &http.Client{
+		Timeout:   defaultHTTPTimeout * time.Second,
+		Transport: newHTTPTransport(),
+	}
+}
+
+// newHTTPTransport returns a reference to an initialized and configured HTTP transport.
+func newHTTPTransport() *http.Transport {
+	return &http.Transport{
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			Timeout:   defaultDialTimeout * time.Second,
+			KeepAlive: defaultKeepaliveTimeout * time.Second,
+		}).DialContext,
+		MaxIdleConns:          defaultMaxIdleConns,
+		IdleConnTimeout:       defaultIdleConnTimeout * time.Second,
+		TLSHandshakeTimeout:   defaultTLSHandshakeTimeout * time.Second,
+		ExpectContinueTimeout: defaultExpectContinueTimeout * time.Second,
+	}
+}
+
+// DoRequest performs the HTTP request with the current ServiceClient's HTTPClient.
+// Authentication and optional headers will be added automatically.
+func (client *ServiceClient) DoRequest(ctx context.Context, method, path string, body io.Reader) (*ResponseResult, error) {
+	// Prepare an HTTP request with the provided context.
+	request, err := http.NewRequest(method, path, body)
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Set("User-Agent", client.UserAgent)
+	request.Header.Set("X-Auth-Token", client.TokenID)
+	if body != nil {
+		request.Header.Set("Content-Type", "application/json")
+	}
+	request = request.WithContext(ctx)
+
+	// Send the HTTP request and populate the ResponseResult.
+	response, err := client.HTTPClient.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	responseResult := &ResponseResult{
+		Response:    response,
+		ErrNotFound: nil,
+		ErrGeneric:  nil,
+		Err:         nil,
+	}
+
+	// Check status code and populate custom error body with extended error message if it's possible.
+	if response.StatusCode >= 400 && response.StatusCode <= 599 {
```

Comment:
Let's use constants from standard `http` package instead of just integers, e.x: `http.StatusOK`, `http.StatusNotFound` and so on.

https://pkg.go.dev/net/http?tab=doc#pkg-constants